helpers/crearMovimientoDetalle.go

```diff
@@ -0,0 +1,311 @@
+		logs.Error(err)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - json.Marshal(nuevoDetalleCuenPre)", err, "400")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	registroMovimientoDetalleRespuesta = models.MovimientoDetalle{
+		Activo:                     true,
+		Descripcion:                "Creación de movimiento detalle",
+		Detalle:                    string(nuevoDetalleCuenPreCast),
+		MovimientoProcesoExternoId: &registroMovProcExterno,
+		Saldo:                      nuevoSaldo,
+		Valor:                      nuevoValor,
+	}
+
+	return registroMovimientoDetalleRespuesta, nil
+}
+
+func CalcularMontos(detalleCuenPre string, saldo float64, valor float64, publicar bool) (detalAcumRespuesta float64, saldoRespuesta float64, valorRespuesta float64, outputError map[string]interface{}) {
+	defer errorctrl.ErrorControlFunction("CalcularMontos - Unhandled Error!", "500")
+
+	cuentaSolicitada := models.CuentasMovimientoProcesoExterno{
+		Cuen_Pre: detalleCuenPre,
+	}
+
+	var detalleUltimoMovimientoDetalle map[string]interface{}
+	var result models.MovimientoDetalle
+	var err map[string]interface{}
+
+	if result, err = GetUltimo(cuentaSolicitada); err != nil {
+		logs.Warn(err)
+		if saldo != 0 {
+			return saldo, saldo, saldo, nil
+		} else if valor != 0 {
+			return valor, valor, valor, nil
+		} else {
+			outputError := errorctrl.Error("RegistroMovimientoDetalle - CalcularMontos(detalleCuenPre)", err, "500")
+			return 0, 0, 0, outputError
+		}
+	} else if err == nil && publicar {
+		if saldo != 0 {
+			return 0, saldo, saldo, nil
+		} else if valor != 0 {
+			return 0, valor, valor, nil
+		} else {
+			outputError := errorctrl.Error("RegistroMovimientoDetalle - CalcularMontos(detalleCuenPre)", err, "500")
+			return 0, 0, 0, outputError
+		}
+	}
+
+	// logs.Debug("RESULTADO GET ULTIMO: ", result)
+
+	if err := json.Unmarshal([]byte(result.Detalle), &detalleUltimoMovimientoDetalle); err != nil {
+		logs.Error(err)
+		outputError := errorctrl.Error("CalcularMontos - json.Unmarshal([]byte(result.Detalle), &detalleUltimoMovimientoDetalle)", err, "400")
+		return 0, 0, 0, outputError
+	}
+
+	switch detalleUltimoMovimientoDetalle["DeltaAcum"].(type) {
+	case nil:
+		if saldo != 0 {
+			return saldo, saldo, saldo, nil
+		} else if valor != 0 {
+			return valor, valor, valor, nil
+		}
+	}
+
+	if saldo != 0 {
+		saldoRespuesta = saldo
+		valorRespuesta = saldo - detalleUltimoMovimientoDetalle["DeltaAcum"].(float64)
+		detalAcumRespuesta = valorRespuesta + result.Saldo
+	} else if valor != 0 {
+		valorRespuesta = valor
+		saldoRespuesta = detalleUltimoMovimientoDetalle["DeltaAcum"].(float64) + valor
+		detalAcumRespuesta = valor + result.Saldo
+	}
+
+	return detalAcumRespuesta, saldoRespuesta, valorRespuesta, nil
+
+}
+
+func CrearMovimientosDetalle(cuentasMovimientoDetalle []models.CuentasMovimientoProcesoExterno, publicar bool) (cuentasMovimientoDetalleRespuesta []models.MovimientoDetalle, outputError map[string]interface{}) {
```

Comment:
1. Linea muy larga
2. Falta documentar, cambiar el `...` del _suggested change_ según sea adecuado
```suggestion
// CrearMovimientosDetalle
func CrearMovimientosDetalle(cuentasMovimientoDetalle []models.CuentasMovimientoProcesoExterno,
publicar bool) (cuentasMovimientoDetalleRespuesta []models.MovimientoDetalle,
outputError map[string]interface{}) {
```